lesson-15/src/index.js

```diff
@@ -0,0 +1,140 @@
+(async () => {
+  async function getCategories() {
+    const response = await fetch('https://dummyjson.com/products/categories')
+    const categories = await response.json()
+    return categories
+  }
+
+  console.log(await getCategories())
+
+  const listOfCategories = await getCategories()
+
+  console.log(listOfCategories.length)
+
+  function findCategoryByName(arrayOfCategories, nameOfCategory) {
+    return arrayOfCategories.find((category) => category.name === nameOfCategory)
+  }
+
+  console.log(findCategoryByName(listOfCategories, 'Groceries'))
+  console.log(findCategoryByName(listOfCategories, 'Electronics'))
+
+  // 4
+  async function getCategoryProducts(arrayOfCategories, nameOfCategory) {
+    const category = arrayOfCategories.find((categoryItem) => categoryItem.name === nameOfCategory)
+    if (category !== undefined) {
+      const response = await fetch(category.url)
+      const listOfProducts = await response.json()
+      return listOfProducts.products.map((product) => ({
+        id: product.id,
+        title: product.title
+      }))
+    } return `категория ${nameOfCategory} не найдена`
+  }
+
+  getCategoryProducts(listOfCategories, 'Electronics').then((result) => {
+    console.log(result)
+  })
+
+  getCategoryProducts(listOfCategories, 'Groceries').then((result) => {
+    console.log(result)
+  })
+
+  // 5
+  async function fetchProductsNamesAndPrises(arrayOfCategories, nameOfCategory) {
+    const category = arrayOfCategories.find((categoryItem) => categoryItem.name === nameOfCategory)
+    if (category !== undefined) {
+      const response = await fetch(category.url)
+      const listOfProducts = await response.json()
+      return Object.fromEntries(listOfProducts.products.map((product) => [product.title, product.price]))
+    } return `категория ${nameOfCategory} не найдена`
+  }
+
+  fetchProductsNamesAndPrises(listOfCategories, 'Electronics').then((result) => {
+    console.log(result)
+  })
+
+  fetchProductsNamesAndPrises(listOfCategories, 'Groceries').then((result) => {
+    console.log(result)
+  })
```

Comment:
```suggestion
  async function fetchProductsNamesAndPrices(categories, categoryName) {
    const category = categories.find(({ name }) => name === categoryName)

    if (category) {
      const response = await fetch(category.url)
      const { products } = await response.json()

      return Object.fromEntries(products.map(({ price, title }) => [title, price]))
    }

    return `категория ${categoryName} не найдена`
  }

  console.log(await fetchProductsNamesAndPrices(listOfCategories, 'Electronics'))
  console.log(await fetchProductsNamesAndPrices(listOfCategories, 'Groceries'))
```

чуть зарефакторил с деструктуризацией

- пофиксил ошибку в названии ф-ции
- сократил названия параметров и переменных
- добавил деструктуризацию
- обновил if
- убрал then
- добавил пустые строчки чтобы для читаемости